internal/web/server.go

```diff
@@ -256,15 +264,37 @@ func (s *server) Start(ctx context.Context) error {
 		_ = cliutils.OpenInBrowser("http://" + bindAddr)
 	}
 
-	go s.broadcaster.Run()
-	server := &http.Server{}
-	err = server.Serve(s.listener)
+	go s.broadcaster.Run(s.stopCh)
+	s.httpServer = &http.Server{}
+
+	var receivedSig *os.Signal
+	go func() {
+		sigint := make(chan os.Signal, 1)
+		signal.Notify(sigint, syscall.SIGTERM, syscall.SIGINT)
+		sig := <-sigint
+		receivedSig = &sig
+		s.shutdown()
+	}()
+
+	err = s.httpServer.Serve(s.listener)
 	if err != nil && err != http.ErrServerClosed {
 		return err
 	}
+
+	<-s.httpServerHasShutdownCh
+	cliutils.ExitFromSignal(receivedSig)
+
 	return nil
 }
 
+func (s *server) shutdown() {
+	close(s.stopCh)
+	if err := s.httpServer.Shutdown(context.Background()); err != nil {
```

Comment:
Since shutdown may wait indefinitely until all connections are closed, maybe it would be a good idea to pass a context with timeout here? Something around 10s?

Reply:
Good idea :+1: